File: src/main/java/pl/yalgrin/playnite/simplesync/web/ChangeResource.java
package pl.yalgrin.playnite.simplesync.web;


import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import pl.yalgrin.playnite.simplesync.dto.ChangeDTO;
import pl.yalgrin.playnite.simplesync.dto.GameChangeRequestDTO;
import pl.yalgrin.playnite.simplesync.service.ChangeService;
import reactor.core.publisher.Flux;

@RestController
@RequestMapping("/api/change")
@RequiredArgsConstructor
public class ChangeResource {

    private final ChangeService service;

    @GetMapping
    public Flux<ChangeDTO> fetchChanges(@RequestParam(required = false) Long lastChangeId) {
        return service.findFromLastId(lastChangeId);
    }

    @GetMapping(value = "/all")
    public Flux<ChangeDTO> fetchAllObjects() {
        return service.generateChangesForAllObjects();
    }

    @PostMapping(value = "/games")
    public Flux<ChangeDTO> fetchSelectedGameChanges(@RequestBody GameChangeRequestDTO dto) {
        return service.generateChangesForGames(dto);
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/web/change/ChangeResource.kt
package pl.yalgrin.playnite.simplesync.web.change

import org.springframework.web.bind.annotation.*
import pl.yalgrin.playnite.simplesync.change.dto.GameChangeRequestDTO
import pl.yalgrin.playnite.simplesync.change.service.ChangeService
import pl.yalgrin.playnite.simplesync.client.message.ChangeMessage
import reactor.core.publisher.Flux

@RestController
@RequestMapping("/api/change")
class ChangeResource(
    val service: ChangeService
) {

    @GetMapping
    fun fetchChanges(@RequestParam(required = false) lastChangeId: Long?): Flux<ChangeMessage> {
        return service.findFromLastId(lastChangeId)
    }

    @GetMapping("/all")
    fun fetchAllObjects(): Flux<ChangeMessage> {
        return service.generateChangesForAllObjects()
    }

    @PostMapping("/games")
    fun fetchSelectedGameChanges(@RequestBody dto: GameChangeRequestDTO): Flux<ChangeMessage> {
        return service.generateChangesForGames(dto)
    }
}